src/containers/ImageUploader/EditImage.tsx

```diff
@@ -6,17 +6,72 @@
  */
 
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
+import { connect, ConnectedProps } from 'react-redux';
 import { withRouter } from 'react-router-dom';
 
+import { RouteComponentProps } from 'react-router';
 import { actions as licenseActions, getAllLicenses } from '../../modules/license/license';
 import { getLocale } from '../../modules/locale/locale';
 import ImageForm from './components/ImageForm';
 import { actions, getImage } from '../../modules/image/image';
-import { ImageShape } from '../../shapes';
+import { UpdatedImageMetadata } from '../../modules/image/imageApiInterfaces';
+import { License, ReduxState } from '../../interfaces';
 
-class EditImage extends Component {
+interface ImageType extends UpdatedImageMetadata {
+  revision?: number;
+  imageFile?: string | Blob;
+}
+
+interface DispatchTypes {
+  fetchLicenses: () => void;
+  fetchImage: (imageInfo: { id: string; language?: string }) => void;
+  updateImage: (imageInfo: {
+    image: ImageType;
+    file: string | Blob;
+    history: RouteComponentProps['history'];
+    editingArticle?: boolean;
+  }) => void;
```

Comment:
Har sett litt på koden men klarer ikke helt å skjønne, er dette redux-actions sine funksjoner eller er det funksjoner fra draft-api/image-api? For hvis det er det andre så må return-typen endres til `Promise<>`

Reply:
Det er redux-actions ja. Så `fetchImage` er en redux-action som henter bildet fra redux-staten (også er det en annen action som oppdaterer det). Så tror ikke vi trenger noen promises her :)

Reply:
De mappes inn med connect saken (via `mapDispatchToProps` objektet)